File: lab2/src/List/LinkedListFactory.scala
package List

import List.IUserType
import objects.{MyDot, MyVector}

class LinkedListFactory {
  private var list = Array(new MyVector, new MyDot)
  private var linkedList = new LinkedList[IUserType]()
  private var listClass = list(1)

  private def findListClass(className: String): IUserType = {
    for (classItem <- list) {
      if (classItem.getClassName == className) {
        listClass = classItem
        return listClass
      }
    }
    listClass = null
    listClass
  }

  def getBuilderByName(className: String):  LinkedList[IUserType] = {
    val listtype = findListClass(className)
    val clone = linkedList.copy
    clone.setType(listtype)
    clone
  }

  def getList: Array[String] = {
    val items = new Array[String](list.length)
    for (i <- 0 until list.length) {
      items(i) = list(i).getClassName
    }
    items
  }
}

File: lab3/src/main/kotlin/List/LinkedListFactory.kt
package List

import Objects.MyDot

import Objects.MyVector


class LinkedListFactory {
    private val list = arrayOf<IUserType>(MyVector(), MyDot())
    private val linkedList: LinkedList<IUserType> = LinkedList<IUserType>()
    private var listClass: IUserType? = null
    private fun findListClass(className: String): IUserType? {
        for (classItem in list) {
            if (classItem.getClassName() == className) {
                listClass = classItem
                return listClass
            }
        }
        listClass = null
        return listClass
    }

    fun getBuilderByName(className: String): LinkedList<IUserType?> {
        val type = findListClass(className)
        val clone: LinkedList<IUserType?> = linkedList.copy()
        clone.setType(type!!)
        return clone
    }

    fun getList(): Array<String?> {
        val items = arrayOfNulls<String>(list.size)
        for (i in list.indices) {
            items[i] = list[i].getClassName()
        }
        return items
    }
}